Library/Homebrew/os/mac/keg.rb

```diff
@@ -8,6 +8,12 @@ def change_dylib_id(id, file)
     @require_relocation = true
     odebug "Changing dylib ID of #{file}\n  from #{file.dylib_id}\n    to #{id}"
     MachO::Tools.change_dylib_id(file, id, strict: false)
```

Comment:
I think it would be ideal if this didn't do the code signing but it was done separately with `MachO::Tools.codesign` or similar. That allows us to catch the exception just for the code signing stage and generally separate the two. It'll also make it easier to have e.g. a DSL to enable/disable code-signing per-formula.

Reply:
I'm of two minds about this. Let me explain...

Ideally, I think ruby-macho shouldn't "break" binaries. That is, from the gem's perspective (and not Homebrew's perspective), if we make any changes to a binary it should leave at the same level of correctness as where we started. Today, that's not the case: if a mach-o binary is signed, and we then modify that binary with ruby-macho, we leave it with an invalid code signature and the OS will refuse to run or load it until it gets resigned later. `install_name_tool` and friends automatically apply an ad-hoc code signature when modifying binaries and that's the behaviour I was looking to replicate in ruby-macho.

On the other hand, some of the considerations around codesigning are helped if we have access to Homebrew's knowledge about the host OS and the target OS - things that ruby-macho isn't aware of. Based on that, I can see the argument for having ruby-macho leave the actual call to perform signing up to Homebrew, even if the utility function lives in ruby-macho.

cc @woodruffw - I'd love to get your call here.

Reply:
Yeh, that argument is convincing and I agree with it. I remove my objections, thanks @mistydemeo!

> It'll also make it easier to have e.g. a DSL to enable/disable code-signing per-formula.

I think this would be nice but it does look like (please correct me if I'm wrong!) that `codesign!` is already a public API so we could call it directly if desired.

Reply:
> I think this would be nice but it does look like (please correct me if I'm wrong!) that `codesign!` is already a public API so we could call it directly if desired.

That is correct! It's already a public API and we can use it as desired.